test/modules/aws-ecr-integration.ts

```diff
@@ -1,251 +1,475 @@
+import { execSync } from 'child_process';
+
 import config from '../../src/config';
-import * as iasql from '../../src/services/iasql'
-import { getPrefix, runQuery, runInstall, runUninstall, runApply, finish, execComposeUp, execComposeDown, runSync, } from '../helpers'
+import * as iasql from '../../src/services/iasql';
+import {
+  getPrefix,
+  runQuery,
+  runInstall,
+  runUninstall,
+  runApply,
+  finish,
+  execComposeUp,
+  execComposeDown,
+  runSync,
+} from '../helpers';
 
 const prefix = getPrefix();
 const dbAlias = 'ecrtest';
 const repositoryName = prefix + dbAlias;
+const region = 'us-east-1';
 const pubRepositoryName = `pub${prefix}${dbAlias}-${process.env.AWS_REGION ?? 'barf'}`;
-const policyMock = '{ "Version": "2012-10-17", "Statement": [ { "Sid": "DenyPull", "Effect": "Deny", "Principal": "*", "Action": [ "ecr:BatchGetImage", "ecr:GetDownloadUrlForLayer" ] } ]}';
-const updatePolicyMock = '{ "Version": "2012-10-17", "Statement": [ { "Sid": "DenyPull", "Effect": "Deny", "Principal": "*", "Action": [ "ecr:BatchGetImage" ] } ]}';
+const policyMock =
+  '{ "Version": "2012-10-17", "Statement": [ { "Sid": "DenyPull", "Effect": "Deny", "Principal": "*", "Action": [ "ecr:BatchGetImage", "ecr:GetDownloadUrlForLayer" ] } ]}';
+const updatePolicyMock =
+  '{ "Version": "2012-10-17", "Statement": [ { "Sid": "DenyPull", "Effect": "Deny", "Principal": "*", "Action": [ "ecr:BatchGetImage" ] } ]}';
 const apply = runApply.bind(null, dbAlias);
 const sync = runSync.bind(null, dbAlias);
 const install = runInstall.bind(null, dbAlias);
 const uninstall = runUninstall.bind(null, dbAlias);
 const query = runQuery.bind(null, dbAlias);
 const modules = ['aws_ecr'];
+const dockerImage = 'public.ecr.aws/docker/library/hello-world:latest';
+const repositoryTag = 'v1';
 
 jest.setTimeout(240000);
-beforeAll(async () => await execComposeUp());
+beforeAll(async () => {
+  // pull sample image
+  execSync(`docker pull ${dockerImage}`);
+
+  await execComposeUp();
+});
 afterAll(async () => await execComposeDown());
 
 describe('ECR Integration Testing', () => {
-  it('creates a new test db', (done) => void iasql.connect(
-    dbAlias,
-    'not-needed', 'not-needed').then(...finish(done)));
+  it('creates a new test db', done =>
+    void iasql.connect(dbAlias, 'not-needed', 'not-needed').then(...finish(done)));
 
   it('installs the aws_account module', install(['aws_account']));
 
-  it('inserts aws credentials', query(`
+  it(
+    'inserts aws credentials',
+    query(
+      `
     INSERT INTO aws_credentials (access_key_id, secret_access_key)
     VALUES ('${process.env.AWS_ACCESS_KEY_ID}', '${process.env.AWS_SECRET_ACCESS_KEY}')
-  `, undefined, false));
+  `,
+      undefined,
+      false,
+    ),
+  );
```

Comment:
the format seems to be changing drastically which is totally fine, but there are so many re-formats that it makes the *actual* changes harder to track. can we avoid this in the future by doing it in separate PRs / commits?

Reply:
(this is a nit)